### ISSUE TITLE
The "Discourse integrates with Mozilla Directory profiles" prompt is displayed after the discourse profile was created

### ISSUE THREAD

**Issue by rleitan** [mozilla/discourse#200]
[STR]:
1.Press the login button in discourse staging
2.On the "Discourse integrates with Mozilla Directory profiles" dialogue prompt - select the checkbox for "privacy policy and terms of service" and click Not right now button
3.On the Create Account dialogue prompt, click "x" button
4.press log in button in discourse staging
5.On the "Discourse integrates with Mozilla Directory profiles" dialogue prompt - select the checkbox and click Not right now button
6.On the Create Account dialogue prompt, click "Create new account" button

[Expected result]:
The "Discourse integrates with Mozilla Directory profiles" dialogue prompt should not be displayed in step 5

[Actual result]:
Something went wrong message is displayed, the user was created when I clicked the "privacy policy and terms of service checkbox in step 2
![2019-05-08_17h57_10](https://user-images.githubusercontent.com/41474886/57389992-07ce4c80-71c4-11e9-8e71-358c31a6f698.png)

Sometimes, Bad Gateway error is displayed instead of Something went wrong

**Comment by LeoMcA** [comment]
I can reproduce the occasional bad gateway errors on staging, but can't reproduce the "something went wrong message" either locally or on staging.

I wonder if we're not giving enough resources to our staging instance and if it's what causing this bug. Creating a user is a relatively intensive process, perhaps all the creation/deletion during QA is causing the server to be overloaded, and produce this weird behaviour.

Jenkins isn't behaving at the moment so I can't change the resource allocation. Will try again tomorrow.

> [Expected result]:
The "Discourse integrates with Mozilla Directory profiles" dialogue prompt should not be displayed in step 5

IMO this should still be displayed, albeit with a successful account creation at the end of the flow, what were you thinking should be displayed instead?

**Comment by rleitan** [comment]
I assumed the profile was created when I checked "privacy policy and terms of service" checkbox in step 2 (even without clicking "Create the account" button- bug) and then the dialogue prompt from step 5 had no value and caused the error. But the cause might be the one you mentioned. 
I retried several times to reproduce the issue, without success. 

I will keep this bug open for the moment because of the Bad Gateway error.
